src/SqlAsyncCollector.cs

```diff
@@ -261,32 +281,40 @@ private static void GenerateDataQueryForMerge(TableInformation table, IEnumerabl
             // If there are duplicate primary keys, we'll need to pick the LAST (most recent) row per primary key.
             foreach (T row in rows.Reverse())
             {
-                // SQL Server allows 900 bytes per primary key, so use that as a baseline
-                var combinedPrimaryKey = new StringBuilder(900 * table.PrimaryKeys.Count());
-
-                // Look up primary key of T. Because we're going in the same order of fields every time,
-                // we can assume that if two rows with the same primary key are in the list, they will collide
-                foreach (PropertyInfo primaryKey in table.PrimaryKeys)
+                // When T is JObject, currently there isn't way to find out the Primary keys, so it'll be null, adding check.
```

Comment:
I'd just remove this comment, it's clear enough now what the if statement is doing and the else already mentions needing to find out how to get primary keys.